Makefile

```diff
@@ -0,0 +1,134 @@
+# PREFIX is the prefix on the targetsystem
+# DESTDIR can be used to prefix ALL paths, e.g., to do a dummy-install in a
+# fake root dir, e.g., for building packages. Users mainly want PREFIX
+
+PREFIX=/usr/local
+DOCDIR=${PREFIX}/share/doc/skogul
+
+GIT_DESCRIBE:=$(shell git describe --always --tag --dirty)
+VERSION_NO=$(shell echo ${GIT_DESCRIBE} | sed s/[v-]//g)
+OS:=$(shell uname -s | tr A-Z a-z)
+ARCH:=$(shell uname -m)
+
+skogul: $(wildcard *.go */*.go */*/*.go)
+	@echo 🤸 go build !
+	@go build -ldflags "-X main.versionNo=$V" -o skogul ./cmd/skogul
+
+docs/skogul.rst: skogul
+	@echo 😽 Generating documentation $@
+	@./skogul -make-man > $@
+
+skogul.1: docs/skogul.rst
+	@echo 🎢 Generating man-file $@
+	@rst2man < $< > $@
+
+notes: docs/NEWS
+	@echo ⛲ Extracting release notes.
+	@./build/release-notes.sh ${GIT_DESCRIBE} > notes
+
+# MAGIC (I hate noisy Make-runs)
+%/:
+	@mkdir -p $@
+
+all: skogul skogul.1 docs/skogul.rst
+
+install: skogul skogul.1 docs/skogul.rst
+	@echo 🙅 Installing
+	@install -D -m 0755 skogul ${DESTDIR}${PREFIX}/bin/skogul
+	@install -D -m 0644 skogul.1 ${DESTDIR}${PREFIX}/share/man/man1/skogul.1
+	@install -D -m 0644 docs/examples/default.json ${DESTDIR}/etc/skogul/default.json
+	@cd docs; \
+	find -type f -exec install -D -m 0644 {} ${DESTDIR}${DOCDIR}/{} \;
+	@install -D -m 0644 README.rst LICENSE -t ${DESTDIR}${DOCDIR}/
+
+
+# Any complaints on this macro-substitution without patches and I introduce m4.
+rpm-prep/SPECS/skogul.spec: build/redhat-skogul.spec.in | rpm-prep/SPECS/
+	@echo  ❕Building spec-file
+	@cat $< | sed "s/xxVxx/${GIT_DESCRIBE}/g; s/xxARCHxx/${ARCH}/g; s/xxVERSION_NOxx/${VERSION_NO}/g" > $@
+	@which dpkg >/dev/null 2>&1 && { echo 🆒 Adding debian-workaround for rpm build; sed -i 's/^BuildReq/\#Debian hack, auto-commented out: BuildReq/g' $@; } || true
+
+# Build RPM. The spec has a blank %prep, so it assumes sources are already
+# available. This isn't perfect, since it creates a tight coupling between
+# Makefile and specfile, but it isn't all that bad either, since it allows
+# building with minimal redundant effort, and without having to commit to
+# git.
+rpm: rpm-prep/SPECS/skogul.spec | rpm-prep/BUILDROOT/
```

Comment:
This fails for me on my MacOS machine with 
```
$ make rpm
make: *** No rule to make target `rpm-prep/SPECS', needed by `rpm-prep/SPECS/skogul.spec'.  Stop.
```

After creating the rpm-prep/SPECS folder manually, I get 
```
$ make rpm
make: *** No rule to make target `rpm-prep/BUILDROOT', needed by `rpm'.  Stop.
```

Reply:
What version of make do you use?

Reply:
```
$ make --version
GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

This program built for i386-apple-darwin11.3.0
```

Reply:
Works better after 238fee6. Although the build dependency hack doesn't get applied since I don't have dpkg either. :'D

Could (should?) we add the hack in case the system _does not have_ yum, instead of _if it has dpkg?_